src/repos/UserRepo.ts

```diff
@@ -75,8 +87,49 @@ const getUsersFollowingPublication = async (pubSlug: string): Promise<User[]> =>
   return users;
 };
 
+/**
+ * Add article to a user's readArticles
+ */
+const appendReadArticle = async (uuid: string, articleID: string): Promise<User> => {
+  const article = await ArticleRepo.getArticleByID(articleID);
+  const user = await UserModel.findOne({ uuid });
+  const checkDuplicates = (prev: boolean, cur: Article) => prev || cur.id === articleID;
+  if (!user.readArticles.reduce(checkDuplicates, false)) {
+    user.readArticles.push(article);
+  }
+  user.save();
+  return user;
+};
+
+/**
+ * Increment shoutouts in user's numShoutouts
+ */
+const incrementShoutouts = async (uuid: string): Promise<User> => {
+  const user = await UserModel.findOne({ uuid });
+  if (user !== null) {
+    user.numShoutouts += 1;
+    user.save();
+  }
+  return user;
+};
+
+/**
+ * Increment number of bookmarks in user's numBookmarkedArticles
+ */
+const incrementBookmarks = async (uuid: string): Promise<User> => {
+  const user = await UserModel.findOne({ uuid });
```

Comment:
You could use `findOneAndUpdate({'uuid': uuid}, {$inc: {'numBookmarkedArticles`: 1}})`. I believe it'd be faster but not exactly sure on the syntax bc mongoose docs are bad. [Here](https://mongoosejs.com/docs/api.html#model_Model.findOneAndUpdate) it is though.

Reply:
I've tested out both solutions and while both work, it seems that both take around 9-15ms to run per bookmark. A con I found of ```findOneAndUpdate``` was that I found were that I had to disable ```findAndModify``` in order to use it because it is deprecated otherwise. I'm having trouble finding what this changes, so I'm kinda worried about using this method.